### PR TITLE
Add "Key::Unknown" handling

### PR DESCRIPTION
**Pull request by barrowsys**
Working on a project. Turns out i need to test that the dialogues output
a certain way. Turns out there's no way to do that because the case of
"not a terminal" is not handled! It's a quick fix, might as well. I'll
be doing the same for console.

just noticed i accidentally RustFmt'd it too. doesn't hurt but feel free to not accept those changes

### CODE REVIEW

**Comment by pksunkara** [comment]
I am not sure how this will help. Could you please provide how to reproduce this issue?

**Comment by barrowsys** [comment]
issue being that when a dialogue is run in a non-interactive environment, it just sits there, constantly receiving "Key::Unknown" but ignoring it, providing no way to get out of the loop. with this, it will return an error instead.

**Comment by pksunkara** [comment]
Yeah, I understood that part. I was asking to know how I can try dialoguer in a non-interactive environment. Can you please provide a small snippet and/or shell interactions to try?

**Comment by barrowsys** [comment]
my code generates a dialogue, and i want to test that the dialogue is correct. i can redirect stdout to a string (to assert against) but without handling Key::Unknown it just sits there looping forever

**Comment by pksunkara** [comment]
Okay, let me rephrase the whole conversation. Please add a test.

**Comment by barrowsys** [comment]
oh ok sure thing
